ddl/ddl_worker.go

```diff
@@ -952,15 +954,21 @@ func (w *worker) runDDLJob(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64,
 	}()
 	if job.IsFinished() {
 		logutil.Logger(w.logCtx).Debug("[ddl] finish DDL job", zap.String("job", job.String()))
-		return
+		return 0, nil
 	}
+
+	if job.IsPaused() || job.IsPausing() {
+		logutil.Logger(w.logCtx).Info("[ddl] DDL job paused", zap.String("job", job.String()))
+		return ver, pauseReorgWorkers(w, d, job)
```

Comment:
Here if job paused or during pausing period, here will return a pause error, but     in caller of runDDLJob func HandleDDLJobTable did not handle the error.

Reply:
If the job has been paused, just wait for GetWaitTimeWhenErrorOccurred() and then retry it.